runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/DoFnOp.java

```diff
@@ -274,6 +282,49 @@ public void open(
     }
   }
 
+  private String toStepName(ExecutableStage executableStage) {
```

Comment:
static? Since this is more like a util function, please move to a util class, e.g. DoFnUtils or something.

Please also add unit tests for this. Please cover the cases that the ExecutableStage starting from PBegin or ending with PEnd.